### DISCUSSION TITLE
Optics update, Mie code calculations on/offline

### DISCUSSION THREAD

**Discussion by tilmes** [NCAR/CARMA_dev#29] [trop_strat CARMA model]
Currently both CARMA and  MAM using lookup tables that were produced by offline (or in the case of CARMA online) Mie calculations based on the specifics of the assume aerosol scheme.
MAM4 is using a Volume mixing approach, where different species are assumed to be internally mixed, and CARMA is using a core-shell approach. 

CARMA has inline code from Pengei's model, with the original code base located here. The code is added to carma_intr.F90:
/glade/u/home/pengfeiy/cesm1_1_1/models/atm/cam/src/physics/carma/models/aerosol_cld/
MAM4 code base of offline Mie calculation is here: 
/glade/work/tilmes/physprop_code

We decided on the following steps for now:
1. Integrate Pengfei's CARMA inline code into the current trop_strat_carma model code
2. Fix some assumptions (e.g., include both dust and bc in the core calculations)
3. In parallel, integrate an online Mie code into the CARMA model and compare the results
4. Discuss next steps on whether to create a new CARMA inline code to produce look-up tables or not

@cbardeen @fvitt Please add comments to this thread on the progress of this work

**Comment by tilmes** [comment]
Adachi2010: [Paper ](https://drive.google.com/drive/u/0/folders/1ilaDL0fjctl8M-bJfr7rX5RtwraioAiS) that explains different optics approaches and their impact.

**Reply by tilmes** [reply]
We also discussed new aerosol topics diagnostics for CARMA that are aligned with those in MAM4, which now include diagnostics to identify the effects of AOD from different aerosol components. However, we strongly recommend to be careful about these diagnostics, since they assume a volume mixing approach to identify the AOD fraction and don't take into account specifics of the core shell approach, so they are somewhat inconsistent with that is calculated for total AOD.

The correct carma code base: /glade/work/fvitt/camdev/carma_trop_strat (from April 20, 2023) does include these updated diagnostics, and includes a bug fix: compared to /glade/work/fvitt/camdev/carma_trop_strat12:
In subroutine coreshell_aero_lw:

               if (trim(specmorph) == 'core') then
                  if (trim(spectype) == 'dust') then
                     coredustmmr(:ncol)    = coredustmmr(:ncol) + specmmr(:ncol,k)
                  end if
                  if (trim(spectype) == 'bc') then
                     corebcmmr(:ncol)    = corebcmmr(:ncol) + specmmr(:ncol,k)
                  end if
 ...
'bc' needed to be replaced with 'black-c'.

**Comment by cbardeen** [comment]
For #2, the assumption that we should fix for the core is the way the refractive index is mixed. For the imaginary part of the refractive index, a volume mixing approach is used, but for the real part a straight average is used. The volume mixing approach should be used in both cases.

Another thing that we might want to consider is to include the change in refractive index of the sulfates with humidty ( or weight percent). Currently a table is generated based on a single index of refraction bsed upon a single weight percent. As RH changes, it adjusts the Mie calcualtion for particle size, but not for a change in refractive index. It should do both.

For the mixed particles, the shell is treated as sulfate and the refractive index is not altered by the amount of organics or sea salt in the shell. This is something that we may want to change, but is not something that can be changed with the current structure of the lookup table that Pengfei defined. For that we might try #3 first and just calculate the core/shell radii and refractive indices assuming volume mixing of the core and the shell sperately in the optics code and call the Mie code interactively. This could be slow, but it should be easier to implement and short test could suggest whether it is worth the effort to generate new lookup tables and help decide what to do for #4.

**Comment by tilmes** [comment]
**Chucks work:**
- Optics tables from Pengfei are merged into trop_strat , which is very specific to the composition of the aerosols.
- Changing composition would need to required changing the tables.

**New changes: /glade/work/bardeenc/ccsm/case/simone_optics_base**
carma_intr.F90 is where  the Mie code lives. Other code changes are needed.
additional changes: mie and wet radius calculation  had to change.
carma_intr.90  (supports the physics interface of CARMA)
- CARMA_CreateOpticsFile  is the top-level routine to identify what optics type to use. It  iterates over the groups, and asks for a new "opticsType". Do mie.true. is for those elements mie code should be calculated. 
- Different optics types include: I_OPTICS_FIXED (old  models);  OPTICS_MIXED_YU2015 -> current trop_strat model.
Not implemented yet: OPTICS_MIXED_CORESHELL: not implemented yet: more  general uses specific core and shell composition; OPTICS_MIXED_VOLUME: homogeneous particles (like MAM4); OPTICS_MIXED_MAXWELL: different numerical (maxwell approach); OPTICS_SULFATE: refractive index changes with weight percent of  sulfate
- Different optics cases call different sub routines, e.g., for  OPTICS_MIXED_YU2015 : defined ranges for the table.  These ranges could  be changed if we need other information
- One file for each bin will be generated  and ncdf will be written out.
- Mie code loops over relative hum, kappa and need to get a wet radius. 270K is just an inital value, dust/bc ratio is calculated. For the real part of the ref-index, an average of bc and dust has  been done. Chuck suggests to  change this to a volume approach. This has to be tested.
- Mie routine uses I_MIERTN_TOON19981 (Brian's routine does core-shell), others are Botteck(fractal routine)
- For pure sulfates, we only iterate over the weight percent. refindex, is fixed, and does not vary with weight percentage. This could be improved.
carma_model_mod.F90
- nrefidx: number of elements per element (currently only set to 1).
- Added information of refractive index. If these should be changed, you need to regenerate this list and the optics tables. (Comment we could read these indexes from a ncdf file in the future)
- only a few elementes, the shell and bc, dust use the refractive index at this point. 
- new optional isShell for the element for future implementation. 

**To generate the new tables you need to:**
- add to namelist: carma_do_optics  (just done once during startup). 

**Result / information of a run with new Mie code.**
- running the Mie code resulted in new files with new names, that will be written into the run directory.
- If running and you need to add the new file names to the namelist (without a directory name) (and add the existing files to the run directory if you do a branch or hybrid run,  or if you don't want to create the files.
- so change  rad_climate files in user_nl_cam

**Possible test cases:**
- Test the new code using YU2015 optics -> check if optics are the same
-  Test updated code from YU2015 but  use Mie calculation for the core assuming volume mixing ratios for bc/dust for the ref-index for the imaginary code
- Test new run with new lookup tables
- Apply to new bin sizes.

**Comment by tilmes** [comment]
@cbardeen could you post your summary on what you changed for mixfix2 and mixfix3 here?

**Comment by cbardeen** [comment]
Three different tests cases have been created to address some of the issues identified in implementing Pengfei's lookup table generation code. Each of these cases build upon the previous case.

1) MixFix - Use volume mixing for both that real and imaginary parts of the refractive index for the core of the mixed aerosol. Previously, the imaginary part was volume mixed, but the real part was an unweighted average.

2) MixFix2 - The shell may include water, but it Pengfei's implementation the refractive index of the shell was the refractive index of H2SO4 and did not vary with the amount of water in the shell. This has been modified to include volume mixing of the water and H2SO4 for the shell refractive index for both the mixed aerosol group and the pure sulfate group. The refractive index for water was taken from the MAM optical properties files. Two no opticsTypes were introduced for these new implementation: I_OPTICS_MIXED_YU_H2O and I_OPTICS_SULFATE.

3) FixMix3 - The imaginary part of the refractive index for sulfate in Pengfei's code was only specified to 3 decimal points, so anything less than 0.001 was rounded down to zero. These smaller values occur in the visible wavelengths, with MAM having values down to 1e-9 similar to OPAC. The refractive index was switched to the MAM values to carry more precision at these wavelengths.

For sulfates, the difference in extinction between Pengfei's tables and the results using MixFix3 are shown in the attached plot:

![sulfate_550_ext_sw_wtp_MixFix3-Old](https://github.com/NCAR/CARMA_dev/assets/22484676/b3e952bb-f69f-48be-b3c1-6f60e2dc7c5c)
